### ISSUE TITLE
destructuring incorrect error shown

### ISSUE THREAD

**Issue by hozefaj** [eslint/eslint#9587]
**Tell us about your environment**

* **ESLint Version:** 4.5.0
* **Node Version:** 6.10.2
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint

**Please show your full configuration:**
```js
{
    "parser": "babel-eslint",
    "plugins": [
        "promise",
        "babel",
        "react"
    ],
    "rules": {
        "indent": [2, 4],
        "quotes": [2, "single"],
        "linebreak-style": [2, "unix"],
        "semi": [2, "always"],
        "no-console": 1,
        "comma-dangle": [1, "always-multiline"],
        "react/display-name": 1,
        "react/jsx-indent": [2, 4],
        "react/jsx-indent-props": [2, 4],
        "no-underscore-dangle": [2, { "allow": ["_PP"] }],
        "arrow-parens": 0,
        "jsx-a11y/anchor-is-valid": [ "error", {
            "components": [ "Link" ],
            "specialLink": [ "to" ]
          }]
    },
    "env": {
        "node": true,
        "browser": true,
        "mocha": true,
        "es6": true
    },
    "extends": ["airbnb", "prettier"],
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "module",
        "ecmaFeatures": {
          "arrowFunctions": true,
          "templateStrings": true,
          "jsx": true,
          "modules": true
        }
    }
}

```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
let schemaId = '6471268022136718498';
if (this.props.contentType !== 'HTML') {
    schemaId = this.props.contentSchemas.schemaId;
}
```

**What did you expect to happen?**
Not expected to get error inside the `if` condition

**What actually happened? Please include the actual, raw output from ESLint.**
Getting error `[eslint] Use object destructuring`. But when I add the code
```
{ schemaId } = this.props.contentSchemas;
``` 
getting JS syntax error. So I am not sure why to do destructuring here.

**Comment by hozefaj** [comment]
@not-an-aardvark added the required info.

**Comment by not-an-aardvark** [comment]
To use object destructuring at the start of an assignment statement, you need to surround the expression in parentheses:

```js
({ schemaId } = this.props.contentSchemas);
```

Otherwise, the code is not valid JavaScript because the `{` gets interpreted as the start of a block.

**Comment by hozefaj** [comment]
Thanks 👍
